pkg/dogstatsd/server.go

```diff
@@ -377,6 +326,69 @@ func NewServer(demultiplexer aggregator.Demultiplexer, serverless bool) (*Server
 			originOptOutEnabled:       config.Datadog.GetBool("dogstatsd_origin_optout_enabled"),
 		},
 	}
+	return s
+}
+
+func (s *Server) Start(demultiplexer aggregator.Demultiplexer) error {
+
+	// TODO: (components) - DI this into Server when Demultiplexer is made into a component
+	s.demultiplexer = demultiplexer
+
+	packetsChannel := make(chan packets.Packets, config.Datadog.GetInt("dogstatsd_queue_size"))
```

Comment:
Is there a reason to move this code (except the one that depends on `demultiplexer`) from  `NewServer` to `Start`? Reading the configuration options makes a lot of sense to be in `NewServer`.

Reply:
These config options could cause an error. If an error occurs in the constructor, the agent will fail to start. 
Since dogstatsd is an optional component, it would make little sense for a dogstatsd configuration error cause the agent to fail to start when `use_dogstatsd: false`. This preserves the existing behavior today since in `main` we do not construct the dogstatsd server until we are ready to start it.

Reply:
If I move the logic back into the constructor - I think I would need to make the constructor infallible and simply log the error instead of return it. Which may cause problems. 

Today - the core agent will continue to run if dogstatsd fails to come up (by ignoring the error)
But standalone dogstasd will halt if the dogstastsd server fails to start (by handling the error)

I could do something like you did in https://github.com/DataDog/datadog-agent/pull/15346  but IMO that's a lot of indirection to expose an error that you could simply handle by calling `fucn Start() err` once the fx app has started. - what do you think?